app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,77 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all.order(:id)
+  end
+
+  def show
+    task_id = params[:id].to_i
+    @task = Task.find_by(id: task_id)
+
+    if @task.nil?
+      redirect_to tasks_path
+    end
+  end
+
+  def new
+    @task = Task.new(name: "Default Name")
+  end
+
+  def create
+    task = Task.new(task_params)
+
+    is_successful = task.save
+
+    if is_successful
+      redirect_to task_path(task.id)
+    else
+      head :not_found
+    end
+  end
+
+  def edit
+    @task = Task.find_by(id: params[:id])
+    if @task.nil?
+      redirect_to tasks_path
+    end
+  end
+
+  def update
+    task = Task.find_by(id: params[:id])
+    task.update(task_params)
+    is_successful = task.save
+
+    if is_successful
+      redirect_to task_path(task.id)
+    else
+      head :not_found
+    end
+  end
+
+  def destroy
+    task = Task.find_by(id: params[:id])
+    if task.nil?
+      head :not_found
+    else
+      task.destroy
+      redirect_to tasks_path
+    end
+  end
+
+  def mark_complete
+    task = Task.find_by(id: params[:id])
```

Comment:
You also should have an if statement to ensure that you find the task before trying to toggle it.